web-app-router/scenes/Portal/Teams/TeamId/Team/ApiKeys/page/ApiKeyTable/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { Footer } from "@/components/Table/Footer";
+import { Header } from "@/components/Table/Header";
+import { FetchKeysQuery } from "../graphql/client/fetch-keys.generated";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { ApiKeyRow } from "./ApiKeyRow";
+import { ViewDetailsModal } from "./ViewDetailsModal";
+import { useResetApiKeyMutation } from "./ApiKeyRow/graphql/client/reset-api-key.generated";
+import { DeleteKeyModal } from "./DeleteKeyModal";
+
+type ApiKeysTableProps = {
+  teamId?: string;
+  apiKeys?: FetchKeysQuery["api_key"];
+};
+
+type ApiKeyRowType = {
+  apiKey: FetchKeysQuery["api_key"][0];
+  openViewDetails: (key: FetchKeysQuery["api_key"][0]) => void;
+};
+
+export const ApiKeysTable = (props: ApiKeysTableProps) => {
+  const { teamId, apiKeys } = props;
+  const rowsPerPageOptions = [10, 20]; // Rows per page options
+  const [currentPage, setCurrentPage] = useState(1);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [selectedKey, setSelectedKey] = useState<
+    FetchKeysQuery["api_key"][0] | null
+  >(null);
+
+  const [showViewDetailsModal, setShowViewDetailsModal] = useState(false);
+  const [showDeleteKeyModal, setShowDeleteKeyModal] = useState(false);
+  const [totalResultsCount, setTotalResultsCount] = useState(
+    apiKeys?.length ?? 0,
+  );
+
+  const openViewDetails = useCallback(
+    (key: FetchKeysQuery["api_key"][0]) => {
+      setSelectedKey(key);
+      setShowViewDetailsModal(true);
+    },
+    [setSelectedKey, setShowViewDetailsModal],
+  );
+
+  const openDeleteKeyModal = useCallback(
+    (key: FetchKeysQuery["api_key"][0]) => {
+      setSelectedKey(key);
+      setShowDeleteKeyModal(true);
+    },
+    [setShowDeleteKeyModal, setSelectedKey],
+  );
+
+  useEffect(() => {
+    setTotalResultsCount(apiKeys?.length ?? 0);
+  }, [apiKeys?.length]);
+
+  const handlePageChange = (newPage: number) => {
+    setCurrentPage(newPage);
+  };
+
+  const handleRowsPerPageChange = (newRowsPerPage: number) => {
+    setRowsPerPage(newRowsPerPage);
+    setCurrentPage(1); // Reset to first page when rows per page changes
+  };
+
+  const headers = [
+    <span key={0}>Name</span>,
+    <span key={1}>API Key</span>,
+    <span key={2}>Created</span>,
+    <span key={3}>Status</span>,
+    null,
+  ];
+
+  const apiKeysToRender = useMemo(() => {
+    if (!apiKeys) {
+      return [];
+    }
+
+    let filteredApiKeys = apiKeys;
+
+    const startIndex = (currentPage - 1) * rowsPerPage;
+    const endIndex = startIndex + rowsPerPage;
+    const paginatedNullifiers = filteredApiKeys.slice(startIndex, endIndex);
+
+    return paginatedNullifiers.map(
+      (apiKey: FetchKeysQuery["api_key"][0]): ApiKeyRowType => {
+        return {
+          openViewDetails: openViewDetails,
+          apiKey: apiKey,
+        };
+      },
+    );
+  }, [apiKeys, currentPage, rowsPerPage]);
+
+  return (
+    <div className="w-full max-h-[380px] overflow-auto no-scrollbar">
```

Comment:
@andy-t-wang I don't think we should limit table height and make it scrollable. It's a bit confusing when you set you've set 10 rows per page but still see only about 5 rows. Also it's not obvious because scrollbar is hidden.